csclient/csclient_test.go

```diff
@@ -1938,3 +2018,39 @@ func (m *termsDischarger) thirdPartyChecker(_ *http.Request, cond, args string)
 	}
 	return nil, &httpbakery.Error{Code: "term agreement required", Message: fmt.Sprintf("term agreement required: %s", strings.Join(needsAgreement, " "))}
 }
+
+type testProgress struct {
+	c *gc.C
+}
+
+func (p testProgress) Start(uploadId string, expires time.Time) {
+	p.c.Logf("progress: start %q %v", uploadId, expires)
```

Comment:
I wonder if, instead of printing testing logs, the Progress used i tests should collect the provided information so that we can run assertions on the collected data.